lms/djangoapps/courseware/tests/test_views.py

```diff
@@ -307,6 +307,55 @@ def test_index_no_visible_section_in_chapter(self):
         self.assertNotIn('Problem 1', response.content)
         self.assertNotIn('Problem 2', response.content)
 
+    def _create_url_for_enroll_staff(self):
+        """
+        User will have satff access and creates the url for enroll_staff view
+        """
+        self.user.is_staff = True
+        self.user.save()  # pylint: disable=no-member
+        self.client.login(username=self.user.username, password=self.password)
+
+        # create the course
+        course = CourseFactory.create()
+
+        # create the _next parameter
+        courseware_url = reverse('courseware', kwargs={
+            'course_id': unicode(course.id)}) + '?activate_block_id=test_block_id'
+
+        # create the url for enroll_staff view
+        enroll_staff_url = "{enroll_staff_url}?next={courseware_url}".format(
+            enroll_staff_url=reverse('enroll_staff', kwargs={'course_id': unicode(course.id)}),
+            courseware_url=courseware_url
+        )
+        return enroll_staff_url
+
+    def test_redirection_unenrolled_staff(self):
+        """
+        Verify unenrolled staff is not redirected to the 'about' section of the chapter
+        """
+        enroll_staff_url = self._create_url_for_enroll_staff()
+
+        # Check the GET method
+        response = self.client.get(enroll_staff_url)
+        self.assertEqual(response.status_code, 200)
+        response_content = response.content
+        self.assertIn('Enroll', response_content)
+        self.assertIn("dont_enroll", response_content)
+
+    @ddt.data(
+        {'enroll': "Enroll"},
+        {'dont_enroll': "Don't enroll"},
+    )
+    def test_redirection_unenrolled_staff_post_data(self, data):
+        """
+        Verify unenrolled staff is redirected to the page according to data passed.
+        """
+        enroll_staff_url = self._create_url_for_enroll_staff()
+        response = self.client.post(enroll_staff_url, data=data)
+
+        # Here we check the status code 302 because of the redirect
+        self.assertEqual(response.status_code, 302)
```

Comment:
If redirected url is different in both scenarios, it seems a good candidate for `@dd.data` :)